packages/core/src/hooks.ts

```diff
@@ -320,7 +320,11 @@ function getAugmentedHooks<T extends Connector>(
   { useChainId, useAccounts }: ReturnType<typeof getStateHooks>,
   { useAccount, useIsActive }: ReturnType<typeof getDerivedHooks>
 ) {
-  function useProvider(network?: Networkish, enabled = true): Web3Provider | undefined {
+  // the default type is set to Web3Provider so that consumers who don't use customProvider aren't impacted
```

Comment:
I'd rephrase this as `// Avoid type erasure by returning the most qualified type if not otherwise set`